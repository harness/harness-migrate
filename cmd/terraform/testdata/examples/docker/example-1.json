{
 "name": "octocat",
 "project": [
  {
   "name": "hello-world",
   "repo": "https://github.com/octocat/hello-world.git",
   "branch": "main",
   "type": "drone",
   "yaml": "a2luZDogcGlwZWxpbmUKdHlwZTogZG9ja2VyCm5hbWU6IGRlZmF1bHQKCnN0ZXBzOgotIG5hbWU6IGhlbGxvCiAgaW1hZ2U6IGJ1c3lib3gKICBjb21tYW5kczoKICAtIGVjaG8gaGVsbG8gd29ybGQKICAKdHJpZ2dlcjoKICBicmFuY2g6CiAgLSBtYWluCg==",
   "secrets": [
    {
     "name": "PROJECT_SECRET",
     "desc": "PROJECT_SECRET",
     "value": "topsecret"
    }
   ]
  }
 ],
 "secrets": [
  {
   "name": "ORG_SECRET",
   "value": "confidential"
  }
 ]
}
