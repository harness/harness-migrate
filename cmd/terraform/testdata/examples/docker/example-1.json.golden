// Locals
locals {
  projects = {
    hello-world = {
      yaml_properties = <<-EOT
        properties:
          ci:
            codebase:
              build: <+input>
              repoName: hello-world
          EOT
      yaml_stages = <<-EOT
        stages:
          - stage:
              identifier: default
              name: default
              spec:
                cloneCodebase: true
                execution:
                  steps:
                  - step:
                      identifier: hello
                      name: hello
                      spec:
                        command: echo hello world
                        connectorRef: exampleDockerConn
                        image: busybox
                      timeout: ""
                      type: Run
                platform:
                  arch: Amd64
                  os: Linux
                runtime:
                  spec: {}
                  type: Cloud
              type: CI
              when:
                condition: <+trigger.branch> == "main"
                pipelineStatus: Success
          EOT
      branch    = "main"
      namespace = "octocat"
      repo      = "hello-world"
      secrets   = {
        PROJECT_SECRET = {
          value = "dG9wc2VjcmV0"
        }
      }
    }
  }
  secrets = {
    ORG_SECRET = {
      value = "Y29uZmlkZW50aWFs"
    }
  }
}

terraform {
  required_providers {
    harness = {
      source  = "harness/harness"
      version = "= 0.19.1"
    }
  }
}

provider "harness" {
  endpoint = "https://app.harness.io/gateway"
  account_id = "DNVIhrzCr9SnPHMQUEvRspB"
}

// Organization
module "organization" {
  source  = "harness-community/structure/harness//modules/organizations"
  version = "~> 0.1"

  name = "exampleOrg"
}
// Organization secrets
module "organization_secrets" {
  for_each = local.secrets

  source  = "harness-community/structure/harness//modules/secrets/text"
  version = "~> 0.1"

  name            = each.key
  organization_id = module.organization.details.id
  value           = base64decode(each.value.value)
}

// Projects
module "projects" {
  for_each = local.projects

  source  = "harness-community/structure/harness//modules/projects"
  version = "~> 0.1"

  name            = each.key
  organization_id = module.organization.details.id
}

// Pipelines
module "pipelines" {
  for_each = local.projects
  
  source  = "harness-community/content/harness//modules/pipelines"
  version = "~> 0.1"

  name            = each.key
  organization_id = module.organization.details.id
  project_id      = module.projects[each.key].details.id
  yaml_data       = <<-EOT
${each.value.yaml_properties}
${each.value.yaml_stages}
EOT
}

// Project secrets
module "project_helloworld_secrets" {
  for_each = local.projects["hello-world"].secrets

  source  = "harness-community/structure/harness//modules/secrets/text"
  version = "~> 0.1"

  name            = each.key
  organization_id = module.organization.details.id
  project_id      = module.projects["hello-world"].details.id
  value           = base64decode(each.value.value)
}
// Pull request trigger
module "trigger_pr" {
  for_each = local.projects
  
  source  = "harness-community/content/harness//modules/triggers"
  version = "~> 0.1"

  name = "Pull Request"
  organization_id = module.organization.details.id
  project_id      = module.projects[each.key].details.id
  pipeline_id     = module.pipelines[each.key].details.id
  yaml_data       = <<-EOT
source:
  type: Webhook
  spec:
    # TODO: support other SCM types
    type: Github
    spec:
      type: PullRequest
      spec:
        connectorRef: 
        autoAbortPreviousExecutions: false
        payloadConditions:
          - key: targetBranch
            operator: Equals
            value: ${each.value.branch}
        headerConditions: []
        repoName: ${each.value.namespace}/${each.value.repo}
        actions:
          - Open
          - Reopen
          - Synchronize
inputYaml: |
  pipeline:
    identifier: ${module.pipelines[each.key].details.id}
    properties:
      ci:
        codebase:
          build:
            type: PR
            spec:
              number: <+trigger.prNumber>
  EOT
}

// Push trigger
module "trigger_push" {
  for_each = local.projects
  
  source  = "harness-community/content/harness//modules/triggers"
  version = "~> 0.1"

  name = "Push"
  organization_id = module.organization.details.id
  project_id      = module.projects[each.key].details.id
  pipeline_id     = module.pipelines[each.key].details.id
  yaml_data       = <<-EOT
source:
  type: Webhook
  spec:
    # TODO: support other SCM types
    type: Github
    spec:
      type: Push
      spec:
        connectorRef: 
        autoAbortPreviousExecutions: false
        payloadConditions:
          - key: targetBranch
            operator: Equals
            value: ${each.value.branch}
        headerConditions: []
        repoName: ${each.value.namespace}/${each.value.repo}
        actions: []
inputYaml: |
  pipeline:
    identifier: ${module.pipelines[each.key].details.id}
    properties:
      ci:
        codebase:
          build:
            type: branch
            spec:
              branch: <+trigger.branch>
  EOT  
}

// Tag trigger
module "trigger_tag" {
  for_each = local.projects
  
  source  = "harness-community/content/harness//modules/triggers"
  version = "~> 0.1"

  name = "Tag"
  organization_id = module.organization.details.id
  project_id      = module.projects[each.key].details.id
  pipeline_id     = module.pipelines[each.key].details.id
  yaml_data       = <<-EOT
source:
  type: Webhook
  spec:
    # TODO: support other SCM types
    type: Github
    spec:
      type: Push
      spec:
        connectorRef: 
        autoAbortPreviousExecutions: false
        payloadConditions:
          - key: <+trigger.payload.ref>
            operator: StartsWith
            value: refs/tags/
        headerConditions: []
        repoName: ${each.value.namespace}/${each.value.repo}
        actions: []
inputYaml: |
  pipeline:
    identifier: ${module.pipelines[each.key].details.id}
    properties:
      ci:
        codebase:
          build:
            type: branch
            spec:
              branch: <+trigger.branch>
  EOT
}